Automattic/jetpack
# Subscription and payment blocks: improve loading animation

Alternative to https://github.com/Automattic/jetpack/pull/37642

It shows a spinning loader when the button is clicked across paid content, donations, and subscription blocks.
 
Note that loading times are long because of being sandboxed. In prod this is faster.

![image](https://github.com/Automattic/jetpack/assets/87168/85c1b826-635e-41e3-b663-63fb69de1d78)

![Screenshot 2024-07-03 at 18 32 09](https://github.com/Automattic/jetpack/assets/87168/e34afc38-3557-40b7-a412-7c51d37d596c)

![Screenshot 2024-07-03 at 15 52 23](https://github.com/Automattic/jetpack/assets/87168/66edd56d-5157-4251-bdcb-6c59048d2adf)

https://github.com/Automattic/jetpack/assets/87168/69553199-c49e-435b-892e-fccb38944d93

Closes the modal before showing the new modal:

https://github.com/Automattic/jetpack/assets/87168/11ee31d5-c203-4f71-855d-85656061e15f


Next step is to improve appearance of the new modal in D150533-code but it needs more work with centering the modal better on small screens.

## Proposed changes:

* Injects spinner into the buttons at client-side
* Shows spinner when button is clicked
* Removes the spinner/loading state before iframe is loaded
* Adds "loaded" event to the form, allowing overlay modal and post subscription modal to wait for the event before hiding those modals. Previously the modals would hide on submit, which was too early.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Add donation block, paid content block and subscription blocks.
* Test them in different themes and visual treatments, and see the loading spinner
* Spinner should fit in different themes/colours/font sizes just fine.
